crate_universe/src/context/crate_context.rs

```diff
@@ -650,23 +667,89 @@ impl CrateContext {
             .iter()
             .flat_map(|target| {
                 target.kind.iter().filter_map(move |kind| {
-                    // Unfortunately, The package graph and resolve graph of cargo metadata have different representations
-                    // for the crate names (resolve graph sanitizes names to match module names) so to get the rest of this
-                    // content to align when rendering, the package target names are always sanitized.
+                    // Unfortunately, The package graph and resolve graph of cargo metadata have
+                    // different representations for the crate names (resolve
+                    // graph sanitizes names to match module names) so to get the rest of this
+                    // content to align when rendering, the package target names are always
+                    // sanitized.
                     let crate_name = sanitize_module_name(&target.name);
 
-                    // Locate the crate's root source file relative to the package root normalized for unix
+                    // Locate the crate's root source file relative to the package root normalized
+                    // for unix
                     let crate_root = pathdiff::diff_paths(&target.src_path, package_root).map(
-                        // Normalize the path so that it always renders the same regardless of platform
+                        // Normalize the path so that it always renders the same regardless of
+                        // platform
                         |root| root.to_string_lossy().replace('\\', "/"),
                     );
+                    let mut srcs = Glob::new_rust_srcs().into();
+                    let mut crate_root = crate_root;
+                    let mut compile_data = None;
+                    let local_patch = package.id.repr.contains("(path+file://");
+                    let temp_components = std::env::temp_dir().components().count() + 1;
+                    let real_root: PathBuf =
+                        package_root.components().skip(temp_components).collect();
+                    if local_patch && !real_root.as_os_str().is_empty() {
+                        let root = real_root.display();
+                        let package = if let Some(workspace) = &workspace.workspace_prefix {
+                            format!("{workspace}/{}", root)
+                        } else {
+                            root.to_string()
+                        };
+                        println!("There's a patch crate at '//{package}'."); 
+                        println!("Make sure that the following is the contents of '//{package}/BUILD.bazel':");
```

Comment:
What's with all the printing?

Reply:
For each crate that's patched out, i'm currently just printing out the contents of the BUILD file that the user needs to place inside the actual crate so that the generated BUILD file in the vendor directory can refer to it. While testing it I'd manually copy those into the appropriate locations upon each run. Like I mentioned in the PR description there's a couple options here:

- cargo-bazel could write those files into the correct locations itself, which is maybe undesirable since it's a bit unexpected that it would be writing generated files to places other than the vendor directory
- it can check that the expected generated files exist at the correct location and print out the contents that should go there if they're not there. That way, upon adding a new patch you'd be notified upon re-vendoring that you need to add the autogenerated BUILD file, but it wouldn't be noise on every run like the current behavior.
- it can silently assume that the user knows what filegroups they need to expose from their patch crates. This relies on the user reading the examples/docs, and the error if they don't do it would be moderately helpful since it'll say something along the lines of "Can't find the filegroup `crate_root` in `//your/patched/crates/foo`

Reply:
I see, this feels really weird, I would not expect the tool to generate anything for a crate that's patched with a path patch. If *anything* I would expect a BUILD file with an `alias`. But needing to create `filegroup`s feels confusing and brittle. Can we eliminate this requirement?